src/libs/actions/Report.js

```diff
@@ -257,14 +257,21 @@ function updateReportWithNewAction(reportID, reportAction) {
     // Always merge the reportID into Onyx
     // If the report doesn't exist in Onyx yet, then all the rest of the data will be filled out
     // by handleReportChanged
-    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+    const updatedAction = {
         reportID,
         unreadActionCount: newMaxSequenceNumber - (lastReadSequenceNumbers[reportID] || 0),
         maxSequenceNumber: reportAction.sequenceNumber,
-        lastMessageTimestamp: reportAction.timestamp,
-        lastMessageText: messageText,
-        lastActorEmail: reportAction.actorEmail,
-    });
+    };
+
+    // If the report action from pusher is a higher sequence number than we know about (meaning it has come from
+    // a chat participant in another application), then the last message text and author needs to be updated as well
+    if (newMaxSequenceNumber > (lastReadSequenceNumbers[reportID] || 0)) {
```

Comment:
NAB, we could maybe do a 

```js
const lastReadSequenceNumber = lastReadSequenceNumbers[reportID] || 0;
```
above the `const updatedAction = {` line and use it here and when calculating the `unreadActionCount`